VMware.CloudFoundation.CertificateManagement.psm1

```diff
@@ -1319,6 +1319,518 @@ Function Install-EsxiCertificate {
+                        Write-Output "Configuration of the Microsoft Certificate Authority in SDDC Manager using ($($certAuthorityUser)): SUCCESSFUL."
+                    } else {
+                        Write-Warning "Configuration of the Microsoft Certificate Authority in SDDC Manager using ($($certAuthorityUser)), already exists: SKIPPED."
+                    }
+                    Write-Output "Configuration a Microsoft Certificate Authority in SDDC Manager completed."
+                } Catch {
+                    $ErrorMessage = $_.Exception.Message
+                    Write-Output "Error was: $ErrorMessage."
+                }
+            } else {
+                Write-Error "Unable to authenticate to SDDC Manager ($($server)): PRE_VALIDATION_FAILED."
+            }
+        } else {
+            Write-Error "Unable to connect to Microsoft Certificate Authority ($certAuthorityFqdn)."
+        }
+    } else {
+        Write-Error "Unable to connect to SDDC Manager ($($server)): PRE_VALIDATION_FAILED."
+    }
+}
+
+Function gatherSddcInventory {
+    Param (
+        [Parameter (Mandatory = $true)] $domainType,
+        [Parameter (Mandatory = $true)] $workloadDomain
+    )
+
+    # Gathers deployment details from SDDC Manager.
+    $sddcMgr = Get-VCFManager
+    $sddcMgrVersion = $sddcMgr.version.split(".")[0]
+
+    $resourcesObject = @()
+    
+    # SDDC Manager
+    if ($domainType -eq "Management") {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $sddcMgr.fqdn
+            'name'       = $sddcMgr.fqdn.split(".")[0]
+            'resourceId' = $sddcMgr.id
+            'type'       = "SDDC_MANAGER"
+        }
+    }
+
+    # vRealize Suite Lifecycle Manager
+    if ($domainType -eq "Management") {
+        $vrslcmNode = Get-VCFvRSLCM
+        if ($vrslcmNode.id -ne "") {
+            $resourcesObject += [pscustomobject]@{
+                'fqdn'       = $vrslcmNode.fqdn
+                'name'       = $vrslcmNode.fqdn.split(".")[0]
+                'resourceId' = $vrslcmNode.id
+                'type'       = "VRSLCM"
+            }
+        }
+    }
+
+    # vCenter Server
+    if (([float]$sddcMgrVersion -ge 4) -AND ($domainType -eq "Management")) {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.type -eq "MANAGEMENT" }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    } else {
+        $domain = Get-VCFWorkloadDomain | Where-Object { $_.name -eq $workloadDomain }
+        $vCenterServer = Get-VCFvCenter | Where-Object { $_.domain.id -eq $domain.id }
+    }
+
+    foreach ($vCenter in $vCenterServer) {
+        $resourcesObject += [pscustomobject]@{
+            'fqdn'       = $vCenter.fqdn
+            'name'       = $vCenter.fqdn.split(".")[0]
+            'resourceId' = $vCenter.id
+            'type'       = "VCENTER"
+        }
+    }
+
+    # NSX
+    if (([float]$sddcMgrVersion -ge 4) -OR ($isLegacyWldnsxT)) {
```

Comment:
I don't thing ` -OR ($isLegacyWldnsxT)` is required here as we're not picking this up from migration and `$isLegacyWldnsxT` is not defined.

Reply:
addressed in (4b8076f22e98c01f68d6ae27749056d4ccb5b0f6)